lib/ui/common/RegExInputFormatter.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/services.dart';
+
+class RegExInputFormatter implements TextInputFormatter {
+  final RegExp _regExp;
+
+  RegExInputFormatter._(this._regExp);
+
+  factory RegExInputFormatter.withRegex(String regexString) {
+    try {
+      final regex = RegExp(regexString);
+      return RegExInputFormatter._(regex);
+    } catch (e) {
+      // Something not right with regex string.
+      assert(false, e.toString());
+      return null;
+    }
+  }
+
+  @override
+  TextEditingValue formatEditUpdate(
```

Comment:
naming - have no idea what formatEditUpdate means. may. just say what it does   "validValue"

Reply:
It couldn't be renamed because is an override of a function. RegExInputFormatter is implementing TextInputFormatter interface